src/DynamoRevit/DynamoRevit.cs

```diff
@@ -477,14 +477,42 @@ private static RevitDynamoModel InitializeCoreModel(DynamoRevitCommandData comma
         internal static Version PreloadAsmFromRevit()
         {
             var asmLocation = AppDomain.CurrentDomain.BaseDirectory;
-            Version libGversion = findRevitASMVersion(asmLocation);
+            Version libGVersion = findRevitASMVersion(asmLocation);
             var dynCorePath = DynamoRevitApp.DynamoCorePath;
             // Get the corresponding libG preloader location for the target ASM loading version.
             // If there is exact match preloader version to the target ASM version, use it, 
             // otherwise use the closest below.
-            var preloaderLocation = DynamoShapeManager.Utilities.GetLibGPreloaderLocation(libGversion, dynCorePath);
+            var preloaderLocation = DynamoShapeManager.Utilities.GetLibGPreloaderLocation(libGVersion, dynCorePath);
+
+            // [Tech Debt] (Will refactor the code later)
+            // The LibG version maybe different in Dynamo and Revit, using the one which is in Dynamo.
+            Version preLoadLibGVersion = PreloadLibGVersion(preloaderLocation);
             DynamoShapeManager.Utilities.PreloadAsmFromPath(preloaderLocation, asmLocation);
-            return libGversion;
+            return preLoadLibGVersion;
+        }
+
+        // [Tech Debt] (Will refactor the code later)
+        /// <summary>
+        /// Return the preload version of LibG.
+        /// </summary>
+        /// <param name="preloaderLocation"></param>
+        /// <returns></returns>
+        internal static Version PreloadLibGVersion(string preloaderLocation)
```

Comment:
@mjkkirschner @aparajit-pratap I feel this tech debt is good for now due to the nature of the bug, of course DynamoCore will provide API pretty soon to get the correct libG version based on ASM version and libG preloader available. If there is no objection, let us fix this as part of next DynamoCore release since it is directly related to host integration support.

Reply:
👍

Reply:
@mjkkirschner @aparajit-pratap @AndyDu1985 @ZiyunShang More details for you, I have confirmed with @angelowang that Civil Team dose not have the same bug here because of their implementation of `GetGeometryFactoryPath()` is different than D4R. See: https://git.autodesk.com/Civil3D/Civil3D/blob/master/Source/Dynamo/AcDynamo/GeometryFactoryPathDetector.cs#L38